01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+```{r}
+#| label: check-number-of-counties-2022
+#| error: true
+
+stopifnot(
+  ratio_house_value_households_county |>
+    distinct(year, state, county) |>
+    count(year) |>
+    filter(year == 2022) |>
+    pull(n) == 3143
+)
+```
+
+Further examination shows that there are 5 counties in our metric data that
+are not in the county populations file. 
+```{r}
+#| label: find-extra-counties
+
+
+all_counties <- read_csv(here::here("geographic-crosswalks", "data", "county-populations.csv")) %>%
+  tidylog::filter(year == 2022) %>%
+  select(-year)
+
+(extra_counties <-
+  ratio_house_value_households_county %>%
+  filter(year == 2022) |>
+  distinct(state, county) |>
+  anti_join(all_counties, by = c("state", "county"))
+)
+```
+
+At the same time, there are 3 counties in the county populations file that are
+not present in our metric data:
+
+```{r}
+#| label: find-missing-counties
+
+(
+  missing_counties <- ratio_house_value_households_county %>%
+    filter(year == 2022) |>
+    distinct(state, county) |>
+    anti_join(x = all_counties, y = _, by = c("state", "county"))
+)
+```
+
+
+## Export {#sec-export}
+
+
+```{r}
+#| label: export-data
+
+export_metric <- function(data, file_type) {
+  geography <- ifelse("place" %in% colnames(data), "city", "county")
+  geo_var <- ifelse(geography == "city", "place", "county")
+
+  pop_file <-
+    read_csv(here::here("geographic-crosswalks", "data", paste0(geo_var, "-populations.csv")))
+  data <- data |> left_join(pop_file, by = c("year", "state", geo_var))
+
+
+  if (file_type == "recent") {
+    data <- filter(data, year == max(year)) |> select(-year)
+    file_type <- 2022
+  }
+
+  if (file_type == "subgroup") {
+    data
+  }
+
+  output_path <- here(glue("01_financial-well-being/data/final/households_house_value_race_ethnicity_{file_type}_{geography}.csv"))
+
+
+  data |>
+    relocate(c(state_name, rlang::sym(paste0(geo_var, "_name"))), .after = {{ geo_var }}) |>
+    select(-population) |>
+    write_csv(file = output_path)
+}
+
+export_metric(ratio_house_value_households_county, file_type = "recent")
+export_metric(ratio_house_value_households_county, file_type = "all")
+
+ratio_house_value_households_county_subgroup |>
+  mutate(subgroup_type = "age", .before = "subgroup") |>
+  bind_rows(ratio_house_value_households_county |>
+    mutate(
+      subgroup_type = "all",
+      subgroup = "All"
+    )) |>
+  arrange(year, state, county, subgroup_type, subgroup) |>
+  export_metric(file_type = "subgroup")
+
+
+export_metric(ratio_house_value_households_place, file_type = "recent")
+export_metric(ratio_house_value_households_place, file_type = "all")
+
+ratio_house_value_households_place_subgroup |>
+  mutate(subgroup_type = "age", .before = "subgroup") |>
+  bind_rows(ratio_house_value_households_place |>
+    mutate(
+      subgroup_type = "all",
+      subgroup = "All"
+    )) |>
+  arrange(year, state, place, subgroup_type, subgroup) |>
+  export_metric(file_type = "subgroup")
+```
```

Comment:
With the changes I made (noted in prior comments), this file reproduces the final output.

I noted, however, several things in the final output: 
1. There are some NAs in the quality flags variables that I am unsure is expected based on the code generating such flags earlier, Are these missing values expected? If not, this should be examined again.
2. Several place final csv files including the all_city and the race_ethnicity_subgroup_city files have all NA values for state_name and place_name variables. This was not an issue in the 2022_city file.

Please let me know if you have any questions about any of my comments.

Reply:
In reviewing the 2022 county file:
1. There are still values for 5 "old" CT counties. These should be removed.
2. Also, there should be observations added for all CT planning regions--only 6 are in the final dataset. 
3. For the metric value, if one part of the ratio is NA, should it be NA:0.8 (as it is written now) or just listed as NA?
4. For the data quality flag, the GitHub standards say that the quality flag should be "" if the metric is NA, but these are listed as NA in the final data, Is this right?
5. Even in recent year data files, should there be a year column? If yes, this should be added.

Reply:
In the subgroup files, I believe that the order of the subgroup within counties/places should be alphabetical.

Reply:
1. I removed the 5 "old" CT counties and added the 3 missing planning regions in 2022.
2.  Claudia suggested leaving the data quality flag as currently written since this is something that can be handled in post-processing. 
3. Similarily, I confirmed with Aaron that since he read all files into R when post-processing, which is indifferent between “” and NA in .csvs, it's okay to leave it as NA instead of "".
4. I checked that `data/00_mobility-metrics_recent.csv` doesn't have a `year` column so it's okay that my recent file doesn't have one either!